client/src/views/Graphs/Graph.vue

```diff
@@ -476,6 +477,40 @@
       this.modalDocumentArtifact = artifact;
       this.showModalDocument = true;
     }
+
+    onAddToSubgraph (edge: GraphEdgeInterface): void {
+      const subgraph = _.clone(this.subgraph);
+      const nodes = subgraph.nodes.map(node => node.label);
+      const edges = subgraph.edges.map(edge => edge.source + '///' + edge.target);
+
+      const bgraphSource = this.bgraphInstance.v({ name: edge.source }).run()[0].vertex;
+      const bgraphTarget = this.bgraphInstance.v({ name: edge.target }).run()[0].vertex;
+
+      // If the edge doesn't exist already in the subgraph
+      if (edges.indexOf(bgraphSource.id + '///' + bgraphTarget.id) === -1) {
+        const bgraphEdge = this.bgraphInstance.v({ source_id: bgraphSource.id, target_id: bgraphTarget.id }).run()[0].vertex;
+         // Check if the source and target nodes already exist in the subgraph
+        if (nodes.indexOf(edge.source) === -1) {
```

Comment:
Not important right now but you can use includes instead of findOf === -1

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes